File: 벤치마크/go/primes.go
package main

import (
	"fmt"
	"time"
)

func main() {
	limit := 1000000

	start := time.Now()

	sieve := make([]bool, limit+1)
	for i := 2; i <= limit; i++ {
		sieve[i] = true
	}

	for i := 2; i*i <= limit; i++ {
		if sieve[i] {
			for j := i * i; j <= limit; j += i {
				sieve[j] = false
			}
		}
	}

	count := 0
	for i := 2; i <= limit; i++ {
		if sieve[i] {
			count++
		}
	}

	elapsed := time.Since(start).Seconds() * 1000

	fmt.Printf("primes up to 1000000: %d\n", count)
	fmt.Printf("RESULT:%.3f\n", elapsed)
}

File: 벤치마크/rust/primes.rs
use std::time::Instant;

fn main() {
    let limit = 1_000_000usize;

    let start = Instant::now();

    let mut sieve = vec![true; limit + 1];
    sieve[0] = false;
    sieve[1] = false;

    let mut i = 2;
    while i * i <= limit {
        if sieve[i] {
            let mut j = i * i;
            while j <= limit {
                sieve[j] = false;
                j += i;
            }
        }
        i += 1;
    }

    let count = sieve[2..=limit].iter().filter(|&&x| x).count();

    let ms = start.elapsed().as_secs_f64() * 1000.0;

    println!("primes up to 1000000: {}", count);
    println!("RESULT:{:.3}", ms);
}
